hypothesis-python/src/hypothesis/strategies.py

```diff
@@ -1809,6 +1809,37 @@ def accept(*args, **kwargs):
     return accept
 
 
+@defines_strategy
+@composite
+def subsequence_of(
+    draw,  # type: ignore
+    elements,  # type: SearchStrategy[Ex]
```

Comment:
The signature should be 

```python
def subsequence_of(
     draw,  # type: ignore
     elements,  # type: Sequence[Ex]
     min_size=0,  # type: int
     max_size=None,  # type: int
 ):
```

Note that elements is a sequence, the average_size arguments are deprecated, and min_size should default to zero (#1618).  

You'll also need to use the `check_sample` validator on `elements`, and make sure failure is a hard error rather than a deprecation for a new strategy!